wp-web-push/readme.txt

```diff
@@ -6,19 +6,29 @@ Tested up to: 4.4.1
 License: GPLv2 or later
 License URI: http://www.gnu.org/licenses/gpl-2.0.html
 
-Web Push notifications for your website.
+Send push notifications to your visitors! Uses the W3C Push API.
+
+== Description ==
+This plugin uses the [W3C Push API](https://www.w3.org/TR/push-api/) to allow you to send push notifications to anyone who visits your site and signs up for notifications.
+
+Once you\'ve installed this plugin, anyone visiting your site/app in [browsers that support the W3C Push API](http://caniuse.com/#feat=push-api) will be asked whether he/she wants to receive notifications from your site. You will immediately be able to send push notifications to anyone who consents.
 
 == Installation ==
-1. Install the plugin from the WordPress.org plugin directory
-2. Activate the plugin
-
-To set up GCM (Google Chrome) support, you need to follow the steps outlined [here](https://developer.mozilla.org/en-US/docs/Web/API/Push_API/Using_the_Push_API#Extra_steps_for_Chrome_support):
-
-1. Navigate to the [Google Developers Console](https://console.developers.google.com/) and set up a new project.
-2. Go to your project\'s homepage (ours is at https://console.developers.google.com/project/push-project-978, for example), then
-  1. Select the Enable Google APIs for use in your apps option.
-  2. In the next screen, click Cloud Messaging for Android under the Mobile APIs section.
-  3. Click the Enable API button.
-3. Now you need to make a note of your project number and API key because you\'ll need them later. To find them:
-  1. Project number: click Home on the left; the project number is clearly marked at the top of your project\'s home page.
-  2. API key: click Credentials on the left hand menu; the API key can be found on that screen.
+1. Download and install the plugin from the WordPress.org plugin directory
+1. Activate the plugin through the \"Plugins\" menu in WordPress
+
+Follow these additional [steps for setting up GCM (Google Chrome) support](https://developer.mozilla.org/en-US/docs/Web/API/Push_API/Using_the_Push_API#Extra_steps_for_Chrome_support)
+
+== Frequently Asked Questions ==
+= What browsers support the W3C Push API? =
+[browser support for the W3C Push API](http://caniuse.com/#feat=push-api) currently exists in Firefox, Chrome, and Chrome for Android, with others likely to follow
+
+= What will push notifications look like? =
+That depends on the browser! Each browser will display your notifications somewhat differently, but in general the notifications will look appropriate for the device/OS/browser on which they are displayed
+
+= When / how often will visitors be asked about accepting push notifications? =
+Each browser has its own heuristics for deciding when to ask its user about push notifications from your site. Most of them won\'t ask the first time someone visits your site, and most also implement a \"never ask me again\" response.
```

Comment:
The heuristic is decided by the plugin itself (and can be configured by the admin).